staging/src/k8s.io/kubectl/pkg/cmd/diff/prune.go

```diff
@@ -0,0 +1,236 @@
+/*
```

Comment:
This file mostly looks like copy-pasted functions from somewhere else? I would have expected the logic to be shared rather than repeated, can we re-use the same piece of code that apply does?

Reply:
Thanks for review @apelisse. Yes, you are right. This file is mostly copy-pasted from `apply/prune.go`. If you support the idea moving `apply/prune.go` to some common place, thanks to that unit tests I wrote also cover both of them.

Reply:
Should this live in `cmd/diff`?

Reply:
OK that seems a little different from what we do in the general prune case.